jbmc/src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -210,6 +210,14 @@ bool ci_lazy_methodst::operator()(
     }
   }
 
+  for(const auto &class_name : instantiated_classes)
+  {
+    const irep_idt cprover_validate =
+      id2string(class_name) + ".cproverNondetInitialize:()V";
+    if(symbol_table.symbols.count(cprover_validate))
+      methods_already_populated.insert(cprover_validate);
```

Comment:
Surely you should also actually populate them!

Reply:
I don't have to populate anything - all of these variables are local and are used to produce this `methods_already_populated` set, which is then used to filter out things to not remove (it looks like). Most of them are badly named. For instance this one should be called `methods_to_load` and probably generated by a separate method.
<br><br>
I initially wanted to refactor this function, but since lazy loading V2 development is in full swing, I decided that would be a wasted effort. If my assumption was wrong, I'll create another PR that cleans this up, and leave this one alone for the sake of merging this fix on time.

Reply:
I meant, you should ensure the method actually is converted, not just put it in an "already_populated" set. Doing that will stop it being removed from the symbol table; putting it into the worklist will get the symbol's value (the method body) populated.

Reply:
But the method body is populated. Otherwise it wouldn't fix the test-gen bug, which is what this change was all about. Really, the only problem is with preexisting misleading variable names.

Reply:
Maybe because there is a special case for it in `add_needed_class`? https://github.com/diffblue/cbmc/blob/develop/jbmc/src/java_bytecode/ci_lazy_methods_needed.cpp#L41

Reply:
That would result in its being added to the worklist and converted (and thus retained) in the usual way. Somehow its body is being filled in without going through the `ci_lazy_methodst` mechanism. Last I heard @LAJW was trying to find out why and submit a fix PR (or justify the status quo)

Reply:
Why it works:
After the `typecheck` phase in which the `ci_lazy_methodst` is run, there is a separate `lazy_goto_model.load_all_functions();` which ensures all symbols are loaded so actually loads the method. 

This fix _shouldn't_ be required because, as @romainbrenguier points out, it is already added to the needed methods by `ci_lazy_methods_needed`. The problem (I suppose) is that a return type of an opaque method is not sufficient to cause loading that type. This is certainly wrong for test-gen, which will stub that method and try and create the object of that type. @smowton do you have an oppinion on whether that is correct for CBMC? That is:

```java
class Foo {
  public int field;
}
class Opaque {
  static Foo opaqueMethod() { return null; }
}
class EntryClass {
  static int entryFunction() { 
    return Opaque.opaqueMethod().field;
  }
}
```
Should `Foo` be loaded?